apps/client/src/common/components/input/colour-input/SwatchPicker.tsx

```diff
@@ -0,0 +1,41 @@
+import { useCallback } from 'react';
+import { IoEyedrop } from '@react-icons/all-files/io5/IoEyedrop';
+import Color from 'color';
+
+import PopoverPicker from '../../../../common/components/input/popover-picker/PopoverPicker';
+import { debounce } from '../../../../common/utils/debounce';
+import { cx } from '../../../utils/styleUtils';
+
+import style from './SwatchSelect.module.scss';
+
+interface SwatchPickerProps {
+  color: string;
+  isSelected?: boolean;
+  onChange: (name: string) => void;
+}
+
+export default function SwatchPicker(props: SwatchPickerProps) {
+  const { color, onChange, isSelected } = props;
+
+  const classes = cx([style.swatch, isSelected ? style.selected : null, style.selectable]);
+
+  const iconColor = Color(color).isLight() && isSelected ? '#000000' : '#FFFFFF';
```

Comment:
The tests are correctly catching an issue where the code will crash if the colour is not a valid colour string (ie: a empty string)

This process needs to be caught, similarly to what we do in the`getAccessibleColour` I linked earlier

Reply:
ah, interesting. thanks, added this on the recent push